### ISSUE TITLE
I want to control commands per connection

### ISSUE THREAD

**Issue by RoraxHan** [nodeftpd/nodeftpd#140]
I want to make ftp server that can control the permission per user.
for example, user type1 can upload the file but can not see the list of file and also can't remove the file.
user type2 can see the list of files but can't delete any file.

**Comment by mk-pmb** [comment]
Set up event listeners for the command events you'd like to hook, e.g. `file:retr` and have them log the arguments that have been passed to the even handler. One of the arguments should be an object that has connection details, including the username. You can use this info to make your event handler decide how to respond.

**Comment by RoraxHan** [comment]
@mk-pmb  Can you provide me example code snippet for me?
Can I hook some events something like below?
connection.on('file:retr')

**Comment by RoraxHan** [comment]
argument is not the problem for now.
I tried to hook like above, and then I wrote console.log but the function is not exeucted.
I mean.. hooking is not working.. so I asked you the example if you can.

**Comment by mk-pmb** [comment]
Try hooking the event on the server then, instead of on the connection.

**Comment by mk-pmb** [comment]
Also make sure you use the nodeftpd from this repo, not the ancient one from npm. And check the context (`this`) of the event handler.

**Comment by RoraxHan** [comment]
I've installed this
`npm install ftpd`


and did like below.
I don't know what is correct 'command:blabla' or 'file:blabla'.
So I tried for both.

```
// blabla...
var server = new ftpd.FtpServer( ....);
server.on('client:connected', function(connection) {
  connection.on('file:dele', function(arg1) {
    console.log('file:dele is hooked here');
  })
  connection.on('command:dele', function(arg1) {
    console.log('command:dele is hooked here');
  })
});
server.listen(options.port);
```
but that's not working..

**Comment by mk-pmb** [comment]
> `npm install ftpd`

Yeah, that's the npm version. It's ancient. I updated #115 with how to update.

> I don't know what is correct 'command:blabla' or 'file:blabla'.

To find out which of them are available, clone the repo and search for `emit(`s that have your keyword:

```bash
git clone https://github.com/nodeftpd/nodeftpd/
cd nodeftpd
git grep -nFe '.emit(' | grep stor
git grep -nFe '.emit(' | grep pass
```
